Tasks/AzureKeyVault/operations/azure-rest/AzureServiceClient.ts

```diff
@@ -0,0 +1,167 @@
+import tl = require('vsts-task-lib/task');
+import util = require("util")
+import msRestAzure = require("./azure-arm-common");
+var httpClient = require('vso-node-api/HttpClient');
+var httpCallbackClient = new httpClient.HttpCallbackClient(tl.getVariable("AZURE_HTTP_USER_AGENT"));
+
+export class WebRequest {
+    public method;
+    public uri;
+    public body;
+    public headers;
+}
+
+export class WebResponse {
+    public statusCode;
+    public headers;
+    public body;
+}
+
+export class ApiResult {
+    public error;
+    public result;
+    public request;
+    public response;
+
+    constructor(error, result?, request?, response?) {
+        this.error = error;
+        this.result = result;
+        this.request = request;
+        this.response = response;
+    }
+}
+
+export class AzureError {
+    public code;
+    public message;
+    public statusCode;
+    public details;
+}
+
+export interface ApiCallback {
+    (error: any, result?: any, request?: any, response?: any): void
+}
+
+export function ToError(response: WebResponse): AzureError {
+    var error = new AzureError();
+    error.statusCode = response.statusCode;
+    error.message = response.body
+    if (response.body && response.body.error) {
+        error.code = response.body.error.code;
+        error.message = response.body.error.message;
+        error.details = response.body.error.details;
+    }
+
+    return error;
+}
+
+export class ServiceClient {
+    private credentials: msRestAzure.ApplicationTokenCredentials;
+    private subscriptionId: string;
+    protected apiVersion: string;
+    protected baseUri: string;
+    protected acceptLanguage: string;
+    protected longRunningOperationRetryTimeout: number;
+    protected generateClientRequestId: boolean;
+
+    constructor(credentials: msRestAzure.ApplicationTokenCredentials, subscriptionId: string) {
+        if (!credentials) {
+            throw new Error(tl.loc("CredentialsCannotBeNull"));
+        }
+        if (!subscriptionId) {
+            throw new Error(tl.loc("SubscriptionIdCannotBeNull"));
+        }
+
+        this.credentials = credentials;
+        this.subscriptionId = subscriptionId
+        this.baseUri = this.credentials.armUrl;
+        this.longRunningOperationRetryTimeout = 60; // In minutes
```

Comment:
copied code and Sachin's team will do the refactoring. I will file a bug on them